src/test/java/org/datadog/jmxfetch/TestGCMetrics.java

```diff
@@ -0,0 +1,182 @@
+package org.datadog.jmxfetch;
+
+import static org.datadog.jmxfetch.util.MetricsAssert.assertDomainPresent;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.io.IOException;
+import java.util.*;
+
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.junit.Test;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.datadog.jmxfetch.reporter.ConsoleReporter;
+import org.datadog.jmxfetch.util.server.MisbehavingJMXServer;
+import org.datadog.jmxfetch.util.server.SimpleAppContainer;
+import org.datadog.jmxfetch.util.MetricsAssert;
+
+@Slf4j
+public class TestGCMetrics extends TestCommon {
+
+    @Test
+    public void testJMXDirectBasic() throws Exception {
+        try (final SimpleAppContainer container = new SimpleAppContainer()){
```

Comment:
Was there any specific reason why you decided to leave in `SimpleApp` in addition to misbehaving-jmx-server?